WebView/app/src/main/assets/main.js

```diff
@@ -22,18 +22,36 @@ function sendAndroidMessage() {
 	`${document.getElementById("shortDescription").innerText} `);
 }
 
+
 function getData() {
-	fetch("https://gcoleman799.github.io/Asset-Loader/weather.json").then(function(resp) {
+    // This JSON files is hosted over the web
+   	fetch("https://raw.githubusercontent.com/android/views-widgets-samples/webview/WebView/sampleData/weather.json").then(function(resp) {
 		return resp.json();
 	}).then(function(data) {
 		var form = document.getElementById("location");
 		var currentLocation = form.options[form.selectedIndex].value;
+		console.log(data[currentLocation].description);
 		document.getElementById("title").innerText = form.options[form.selectedIndex].text;
-        document.getElementById("currentTemp").innerText = data[currentLocation].currentTemp;
+        document.getElementById("currentTemp").innerText = `${data[currentLocation].currentTemp}`+ "\xB0 F";
         document.getElementById("shortDescription").innerText = data[currentLocation].description;
         document.getElementById("longDescription").innerText = "Today in " + `${form.options[form.selectedIndex].text}`
-            + " there is a " + `${data[currentLocation].chancePrecip}` + " chance of precipitation and the humidity is "
-            + `${data[currentLocation].humidity}.`;
-        document.getElementById("icon").src = data[currentLocation].icon;
+            + " there is a " + `${data[currentLocation].chancePrecip}` + "% chance of precipitation and the humidity is "
+            + `${data[currentLocation].humidity}` + "%.";
+        document.getElementById("icon").src = getIcon(data[currentLocation].description);
 	})
+}
+
+/* These icons are hosted locally, in the res/drawable folder. However, we can call them using
+ * http(s):// URLs because we have configured AssetLoader in MainActivity. It is desirable to
+ * access the files in this way because it is compatible with the Same-Origin policy.
+ */
+function getIcon(description){
+    switch(description) {
+        case "Rainy":
+            return "https://raw.githubusercontent.com/views-widgets-samples/res/drawable/rain.png";
+        case "Clear Sky":
+            return "https://raw.githubusercontent.com/views-widgets-samples/res/drawable/sunny.png";
+        default:
+            return "https://raw.githubusercontent.com/views-widgets-samples/res/drawable/partly_cloudy.png";
+    }
```

Comment:
Github keeps showing my changes from your other PRs, which makes this a bit difficult to review. How are you merging the upstream `webview` branch into your development branch? Glancing at [the commit log](https://github.com/android/views-widgets-samples/pull/155/commits) I don't see the merge commits I was expecting. This might be my lack of understanding Github (I don't personally use the ["github flow"](https://guides.github.com/introduction/flow/) workflow).

Reply:
Yes, I noticed that and I apologize for the trouble. 

After I land a PR I have been following these steps:
1. run `git fetch upstream webview:webview`
2. switch onto the branch that is still being worked on and run `git rebase webview`

Does this sound right to you?

Reply:
Are you force-pushing (`git push -f`) your commits?

Another thing you could try (might produce better results):

```sh
# one-time:
git checkout name-of-development-branch
git branch --set-upstream-to=webview

# update your branch on top of latest changes:
git fetch upstream webview:webview
git rebase # rebases on top of the the "set-upstream-to" by default

# force push is required if you rebase:
git push -f origin name-of-development-branch
```

Reply:
Ok I will try that, thanks!